recurring_contract/models/recurring_contract.py

```diff
@@ -657,3 +641,33 @@ def _filter_clean_invoices(self, since_date, to_date):
         if to_date:
             invl_search.append(('due_date', '<=', to_date))
         return invl_search
+
+    def _updt_invoices_rc(self, vals):
+        """
+        It updates the invoices of a contract when the contract is updated
+
+        :param vals: the values that are being updated on the contract
+        """
+        self.ensure_one()
+        if "contract_line_ids" in vals or "group_id" in vals:
+            invoices = self.env['account.move'].search([("partner_id", "in", self.mapped("partner_id").ids),
+                                                        ("move_type", "=", "out_invoice"),
+                                                        ("payment_state", "=", "not_paid"),
+                                                        ("invoice_line_ids.product_id.default_code", "in",
+                                                         ["sponsorship", "sponsorship_ldp", "fund_gen"]),
+                                                        ("invoice_line_ids.contract_id", "in", self.ids)
+                                                        ])
+            if invoices:
+                data_invs = dict()
+                for inv in invoices:
+                    line_data = dict()
+                    for line in self.contract_line_ids:
+                        line_data[line.product_id] = {"amt": line.subtotal, "contract": self}
+                    data_invs.update(
+                        inv._build_invoice_data(
+                            inv_lines=line_data,
+                            ref=self.group_id.ref,
+                            pay_mode_id=self.group_id.payment_mode_id
+                        )
+                    )
+                invoices.update_invoices(data_invs)
```

Comment:
The invoices are already connected with the contract. Why do you filter the "sponsorship" products? As we are in the recurring_contract module, we cannot do that, it must be generic.
```suggestion
        if "contract_line_ids" in vals or "group_id" in vals:
            data_invs = dict()
            for contract in self:
                line_data = dict()
                for line in self.contract_line_ids:
                    line_data[line.product_id] = {"amt": line.subtotal, "contract": self}
                for inv in contract.mapped("invoice_line_ids.move_id").filtered(
                        lambda m: m.payment_state == "not_paid"):
                    data_invs.update(
                        inv._build_invoice_data(
                            inv_lines=line_data,
                            ref=contract.group_id.ref,
                            pay_mode_id=contract.group_id.payment_mode_id
                        )
                    )
            if data_invs:
                self.mapped("invoice_line_ids.move_id").update_invoices(data_invs)
```